content/ConnectingToDatabase/ConnectingToDatabase.q.md

```diff
@@ -0,0 +1,666 @@
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+    $host = 'localhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    
+    $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+    try {
+        $conn = PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is the error message?
+
+- Unable to establish a connection.
+
+- Undefined variable: `dsn` on line 11
+
+- Undefined variable: `conn` on line 12
+
+- Uncaught Error: Call to undefined function `PDO()` on line 11
+
+- Uncaught ArgumentCountError: `PDO::__construct()` expects at least 1 parameter, 0 given on line 11
+
+
+/// type=MS, answer=[2,4]
+
+Which statements correctly describe the error?
+
+- The `$conn` variable is an invalid variable name.
+
+- There is no `new` keyword between `=` and `PDO` on line 11.
+
+- The `$dsn` variable is enclosed inside the parentheses `()`.
+
+- On line 11, the statement `$conn = PDO($dsn);` is incorrect.
+
+- There is a semicolon `;` after the statement `$conn = PDO($dsn)`.
+
+:::
+
+
+/// type=CR, answer=[tests/ConnectingToDatabase/MissingNewKeywordTest.php]
+
+Correct the code so that it outputs the string `Successfully connected!`.
+
+```php
+<?php
+    $host = 'localhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    
+    $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+    try {
+        $conn = PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+
+
+:::
+
+/// type=REPL readonly=true
+
+```php
+<?php
+    $host = 'localhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    
+    $dsn = "pgsql:host=;port=;dbname=;user=;password=";
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected!";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }   
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is its output?
+
+- `An error occurred!`
+
+- No output is displayed.
+
+- `Successfully connected!`
+
+- `Unable to establish a connection.`
+
+- `"Unable to establish a connection."`
+
+
+/// type=MS, answer=[2,3]
+
+Which statements correctly describe the error?
+
+- On line 8, the `$dsn` variable is an invalid variable name.
+
+- Missing variables `$host`, `$db`, `$port`, `$username`, and `$password` on line 8.
```

Comment:
I think we need to improve this...

> There are no values assigned to the required `DSN` parameters on line 8.

 or something else that clearly describes the cause of the error.